src/Pin/index.js

```diff
@@ -23,21 +23,25 @@ import { ApplicationInstance } from '../Launch'
 
 // only used during local development
 let unlocked = false
+let contextItems = ['purchase', 'parental']
 
-let submit = pin => {
+let submit = (pin, context) => {
   return new Promise((resolve, reject) => {
-    if (pin.toString() === Settings.get('platform', 'pin', '0000').toString()) {
+    if (
+      pin.toString() === Settings.get('platform', 'pin', '0000').toString() &&
+      contextItems.includes(context)
+    ) {
       unlocked = true
       resolve(unlocked)
     } else {
-      reject('Incorrect pin')
+      reject('Incorrect pin or Incorrect context')
     }
   })
 }
 
-let check = () => {
-  return new Promise(resolve => {
-    resolve(unlocked)
+let check = context => {
+  return new Promise((resolve, reject) => {
+    contextItems.includes(context) ? resolve(unlocked) : reject('Incorrect Context')
```

Comment:
Same goes for this method. The valid context check should be in the public API methods, before reaching the platform check function